src/status_im/contexts/communities/events.cljs

```diff
@@ -218,12 +224,34 @@
 (rf/reg-event-fx :communities/toggle-selected-permission-address
  toggle-selected-permission-address)
 
+(defn toggle-share-all-addresses
+  [{:keys [db]} [community-id]]
+  (let [share-all-addresses? (get-in db [:communities community-id :share-all-addresses?])
+        accounts             (get-in db [:wallet :accounts])
+        sorted-accounts      (sort-by :position (vals accounts))
+        watched-accounts     (remove #(:watch-only? %) sorted-accounts)
```

Comment:
I have the same logic in three different events. Can you provide guidance on how to make it reusable?

Reply:
1